tests/modules/csm/test_csm_distribution.py

```diff
@@ -539,17 +1042,117 @@ def test_calc_rewards_distribution_in_frame_handles_negative_to_distribute():
     ],
 )
 def test_merge_strikes(
-    module: CSOracle,
     acc: dict,
     strikes_in_frame: dict,
     threshold_per_op: dict,
     expected: dict,
 ):
-    distribution = Distribution(module.w3, Mock(), Mock())
+    distribution = Distribution(Mock(csm=Mock()), Mock(), Mock())
     distribution.w3.csm.get_curve_params = Mock(
         side_effect=lambda no_id, _: Mock(strikes_params=threshold_per_op[no_id])
     )
 
     distribution._merge_strikes(acc, strikes_in_frame, frame_blockstamp=Mock())
 
     assert acc == expected
+
+
+@pytest.mark.parametrize(
+    "total_distributed_rewards, total_rebate, total_rewards_to_distribute",
+    [
+        (100, 50, 150),
+        (0, 0, 0),
+        (50, 50, 100),
+    ],
+)
+def tests_validates_correct_distribution(total_distributed_rewards, total_rebate, total_rewards_to_distribute):
+    Distribution.validate_distribution(total_distributed_rewards, total_rebate, total_rewards_to_distribute)
+
+
+@pytest.mark.parametrize(
+    "total_distributed_rewards, total_rebate, total_rewards_to_distribute",
+    [
+        (100, 51, 150),
+        (100, 50, 149),
```

Comment:
How this is different from the first one?

Reply:
Removed